src/components/toolbar/toolbarFieldCategory.js

```diff
@@ -0,0 +1,168 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useProduct, useProductGraphConfig, useProductToolbarQuery } from '../productView/productViewContext';
+import { reduxTypes, storeHooks } from '../../redux';
+import { RHSM_API_QUERY_SET_TYPES } from '../../services/rhsm/rhsmConstants';
+import { Select, SelectPosition } from '../form/select';
+import { graphCardHelpers } from '../graphCard/graphCardHelpers';
+import { translate } from '../i18n/i18n';
+
+/**
+ * Generate select field options from config
+ *
+ * @param {object} options
+ * @param {Function} options.useProductGraphConfig
+ * @returns {Function}
+ */
+const useToolbarFieldOptions = ({ useProductGraphConfig: useAliasProductGraphConfig = useProductGraphConfig } = {}) => {
+  const { filters } = useAliasProductGraphConfig();
+  const options = [];
+
+  if (Array.isArray(filters)) {
+    const updatedFilters = filters
+      ?.map(({ metric, query }) => {
+        const category = query?.[RHSM_API_QUERY_SET_TYPES.CATEGORY];
+
+        if (category !== undefined) {
+          return {
+            title: translate('curiosity-toolbar.label', {
+              context: ['category', (category === '' && 'unspecified') || category]
```

Comment:
required:
confirm `unspecified` shouldn't follow the consistency pattern of `none`